androidenhancedvideoplayer/src/main/java/com/profusion/androidenhancedvideoplayer/components/PlayerControls.kt

```diff
@@ -79,16 +89,22 @@ fun PlayerControls(
                     customization.nextIconContent()
                 }
             }
-            Box(
+            Row(
+                verticalAlignment = Alignment.CenterVertically,
+                horizontalArrangement = Arrangement.End,
                 modifier = Modifier
                     .fillMaxWidth()
                     .absolutePadding(bottom = Dimensions.large, right = Dimensions.large)
             ) {
+                IconButton(
+                    onClick = { isSettingsOpen = !isSettingsOpen },
+                    modifier = Modifier.testTag("SettingsButton")
+                ) {
+                    SettingsIcon()
+                }
```

Comment:
I'd move those to `Settings` composable. For me, it makes more sense for `Settings` to handle the icon and its test tag modified. Then, `isSettingsOpen` can be a local variable from Settings composable to handle onClick/onDismiss (internal state to control its presentation)

Reply:
I don't like this approach because it makes the settings component non-reusable, as it'll always be associated with the settings icon, for example: if you want to use the settings component with another type of button or with an event listener it won't be possible. Also, it's easier to keep a pattern of all the buttons of the bottomBar if they are in the same place

Reply:
Ok! I was with the assumption that it was an ok trade-off but had not considered a possible listener usage with Settings